### PR TITLE
Feature/aa view obligation summary

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
Integration / data access and unit tests for the new / altered handlers. 

Web side of unit tests next.

### CODE REVIEW

File: src/EA.Weee.DataAccess.Tests.Integration/DataAccess/ObligationDataAccessIntegration.cs [hunk 1]
```diff
@@ -640,25 +640,127 @@ public async Task GetObligationComplianceYears_GivenAuthority_ShouldReturnCompli
...
+
+                var scheme2 = ObligatedWeeeIntegrationCommon.CreateScheme(organisation);
+                ObjectInstantiator<Domain.Scheme.Scheme>.SetProperty(s => s.CompetentAuthorityId, authority2.Id, scheme2);
+
+                var scheme3 = ObligatedWeeeIntegrationCommon.CreateScheme(organisation);
+                ObjectInstantiator<Domain.Scheme.Scheme>.SetProperty(s => s.CompetentAuthorityId, authority2.Id, scheme3);
+
+                var obligationScheme12020 = new ObligationScheme(scheme1, 2022);
+                var obligationScheme22020 = new ObligationScheme(scheme2, 2022);
+                var obligationSchemeNotFound = new ObligationScheme(scheme2, 2023);
```
Comment (meglenapnovoselska) [inline src/EA.Weee.DataAccess.Tests.Integration/DataAccess/ObligationDataAccessIntegration.cs:748]: is that not meant to be scheme3?

File: src/EA.Weee.Integration.Tests/Handlers/GetObligationComplianceYearsHandlerTests.cs [hunk 2]
```diff
@@ -84,6 +78,66 @@ public class WhenIGetSchemeObligationYears : GetObligationComplianceYearsHandler
...
+
+            private readonly Because of = () =>
+            {
+                result = Task.Run(async () => await handler.HandleAsync(request)).Result;
+            };
+
+            private readonly It shouldHaveReturnedObligationYears = () =>
+            {
+                result.Should().NotBeNull();
+                result.Count.Should().Be(2);
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Integration.Tests/Handlers/GetObligationComplianceYearsHandlerTests.cs:136]: that should be 3 no? 
2022, 2021, 2020

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                result = Task.Run(async () => await handler.HandleAsync(request)).Result;
+            };
+
+            private readonly It shouldHaveReturnedObligationYears = () =>
+            {
+                result.Should().NotBeNull();
+                result.Count.Should().Be(2);
+                result.Should().Contain(s => s.Id == scheme1.Id);
+                result.Should().Contain(s => s.Id == scheme2.Id);
+                result.Should().OnlyHaveUniqueItems();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs:91]: You could add result.Should().NotContain for the schemeNotMatchingComplianceYear

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+                request = new GetSchemesWithObligation(SystemTime.UtcNow.Year);
+            };
+
+            private readonly Because of = () =>
+            {
+                result = Task.Run(async () => await handler.HandleAsync(request)).Result;
+            };
+
+            private readonly It shouldHaveReturnedObligationYears = () =>
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs:85]: could be updated to say shouldHaveReturnedSchemesWithObligations

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+            public static IntegrationTestSetupBuilder LocalSetup()
+            {
+                var setup = SetupTest(IocApplication.RequestHandler)
+                    .WithIoC()
+                    .WithTestData(true)
+                    .WithInternalUserAccess();
+
+                var authority = Query.GetEaCompetentAuthority();
+                var role = Query.GetAdminRole();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs:110]: do you need that ? the handler has only    authorization.EnsureCanAccessInternalArea(), doesn't check for the admin role?

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/GetSchemesWithObligationHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    using Prsd.Core.Mapper;
+    using RequestHandlers.Admin.Obligations;
+    using RequestHandlers.Security;
+    using Requests.Admin.Obligations;
+    using System.Security;
+    using System.Threading.Tasks;
+    using Weee.Tests.Core;
+    using Xunit;
+
+    public class GetSchemesWithObligationHandlerTests : SimpleUnitTestBase
```
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/GetSchemesWithObligationHandlerTests.cs:20]: can you add **unit** to the file name and class, it's duplicated with the  EA.Weee.Integration.Tests.Handlers GetSchemesWithObligationHandlerTests name

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    using Domain.Scheme;
+    using FluentAssertions;
+    using NUnit.Specifications;
+    using Prsd.Core;
+    using Prsd.Core.Autofac;
+    using Prsd.Core.Mediator;
+    using Requests.Admin.Obligations;
+    using SchemeStatus = Domain.Scheme.SchemeStatus;
+
+    public class GetSchemesWithObligationHandlerTests : IntegrationTestBase
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs:20]: can you update the file name and class name as adding **integration** to them?

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerViewObligationAndEvidenceSummaryTests.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        [Fact]
+        public void ViewObligationAndEvidenceSummaryGet_IsDecoratedWith_HttpGetAttribute()
+        {
+            typeof(ObligationsController).GetMethod("ViewObligationAndEvidenceSummary", 
+                new[] { typeof(int?), typeof(Guid?) }).Should().BeDecoratedWith<HttpGetAttribute>();
+        }
+
+        [Fact]
+        public void ViewObligationAndEvidenceSummaryGet_ObligationComplianceYearsShouldBeRetrieved()
+        {
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerViewObligationAndEvidenceSummaryTests.cs:46]: is that for later or forgotten?

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/GetSchemesWithObligationHandlerTests.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Admin/Obligations/GetSchemesWithObligationHandlerTests.cs]: done...

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs (hunk 7 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs]: done...

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerViewObligationAndEvidenceSummaryTests.cs (hunk 8 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerViewObligationAndEvidenceSummaryTests.cs:264]: These have now been added.

File: src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs (hunk 5 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs]: Updated...

**Review by meglenapnovoselska — APPROVED** [review]

**Pull request merged**